File: flutter17_mobile/lib/models/person.dart
import 'package:json_annotation/json_annotation.dart';

part 'person.g.dart';

@JsonSerializable()
class Person {
  int? id;
  String? firstName;
  String? lastName;
  DateTime? dateOfBirth;

  Person({
    this.id,
    this.firstName,
    this.lastName,
    this.dateOfBirth,
  });

  factory Person.fromJson(Map<String, dynamic> json) => _$PersonFromJson(json);

  Map<String, dynamic> toJson() => _$PersonToJson(this);
}

File: angular17_web/src/app/models/person/person.ts
export class Person {
  id!: number;
  firstName!: string;
  lastName!: string;
  dateOfBirth!: Date;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.firstName = data.firstName;
    this.lastName = data.lastName;
    this.dateOfBirth = data.dateOfBirth ? new Date(data.dateOfBirth) : undefined as any;
  }
}
